### PR TITLE
python3Packages.stim: init at 1.9.0

### PR DESCRIPTION
**Pull request by ChrisPattison**
###### Description of changes

- Add Python package stim (https://github.com/quantumlib/stim)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ChrisPattison** [comment]
The disabled failing test is in an "extra" orchestration utility and uses multiprocess communication, tempfiles, etc. I'm not sure if that is supposed to work in the sandbox, and at any rate all the remaining tests of the main features pass.

As for the pymatching dependency, it proved to be fairly troublesome to package on several fronts. I am hoping to package it next month and then enable the disabled tests here

File: pkgs/development/python-modules/stim/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  ];
+
+  meta = {
+    description = "A tool for high performance simulation and analysis of quantum stabilizer circuits, especially quantum error correction (QEC) circuits.";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ chrispattison ];
+    homepage = "https://github.com/quantumlib/stim";
+  };
+
+  checkImport = [ "stim" ];
```
Comment (risicle) [inline pkgs/development/python-modules/stim/default.nix:54]: `pythonImportsCheck`?

File: pkgs/development/python-modules/stim/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  # Requires various extra deps
+  disabledTestPaths = [
+    # No pymatching
+    "glue/sample/src/sinter/main_test.py"
+    "glue/sample/src/sinter/decoding_test.py"
+    "glue/sample/src/sinter/predict_test.py"
+    "glue/sample/src/sinter/collection_test.py"
+    "glue/sample/src/sinter/collection_work_manager.py"
+    # Fails
+    "glue/sample/src/sinter/worker_test.py"
```
Comment (risicle) [inline pkgs/development/python-modules/stim/default.nix:67]: Note the failing tests from this file are attempting to use `decoder='pymatching'`

File: pkgs/development/python-modules/stim/default.nix (hunk 1 above)
Comment (ChrisPattison) [reply pkgs/development/python-modules/stim/default.nix]: Thanks for the catch!

File: pkgs/development/python-modules/stim/default.nix (hunk 2 above)
Comment (ChrisPattison) [reply pkgs/development/python-modules/stim/default.nix:65]: So it is. I missed this when going through the failure reasons. Comment changed appropriately

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**